RecoEcal/EgammaClusterAlgos/src/EndcapPiZeroDiscriminatorAlgo.cc

```diff
@@ -10,52 +10,34 @@
 
 using namespace std;
 
+namespace {
+  constexpr int Nfiles_EE = 5;
+}
+
 EndcapPiZeroDiscriminatorAlgo::EndcapPiZeroDiscriminatorAlgo(double stripEnergyCut, int nStripCut, const string& path) :
-   preshStripEnergyCut_(stripEnergyCut),  preshSeededNstr_(nStripCut), pathToFiles_(path)
+   preshStripEnergyCut_(stripEnergyCut),  preshSeededNstr_(nStripCut)
 {   
    
      // Read all Weight files
-     Nfiles_EB = 5;
-     Nfiles_EE = 5;
-     string file_pt[5] = {"20","30","40","50","60"};
-     string file_barrel_pt[5] = {"20","30","40","50","60"};   
-
-     string nn_paterns_file  = "";
-     for(int j=0;j<Nfiles_EE;j++) {
-       nn_paterns_file = "endcapPiZeroDiscriminatorWeights_et"+file_pt[j]+".wts";
-       edm::FileInPath WFile(pathToFiles_+nn_paterns_file);
-       readWeightFile(WFile.fullPath().c_str()); // read the weights' file
-
-       EE_Layers = Layers;
-       EE_Indim = Indim;
-       EE_Hidden = Hidden;
-       EE_Outdim = Outdim;
-          
-       for(int i=0;i<Indim*Hidden;i++)  I_H_Weight_all.push_back(I_H_Weight[i]);
-       for(int i=0;i<Hidden;i++)  H_Thresh_all.push_back(H_Thresh[i]);
-       for(int i=0;i<Outdim*Hidden;i++)  H_O_Weight_all.push_back(H_O_Weight[i]);
-       for(int i=0;i<Outdim;i++)  O_Thresh_all.push_back(O_Thresh[i]);
-     }
+     constexpr std::array<char const*,Nfiles_EE> file_pt{ {"20","30","40","50","60"} };
+     constexpr std::array<char const*,5> file_barrel_pt { {"20","30","40","50","60"} };   
```

Comment:
Thank you @Dr15Jones for the update!

However, looking at the code I notice that the size of the two arrays  ```file_pt``` and ```file_barrel_pt``` must necessarily be the same by construction: in fact, both 
```
   float GetNNOutput(float EE_Et);
   float GetBarrelNNOutput(float EB_Et); 
```
have a call to
```
  float getNNoutput(int sel_wfile, int Layers, int Indim, int Hidden, int Outdim, int barrelstart) const;
```
which uses ```Nfiles_EE``` for dimensioning the vector.

At this point, it is certainly cleaner to define both
```
    constexpr std::array<char const*,Nfiles_EE> file_pt{ {"20","30","40","50","60"} };
    constexpr std::array<char const*,Nfiles_EE> file_barrel_pt { {"20","30","40","50","60"} }; 
```
(and, maybe even better, rename ```Nfiles_EE``` ->```Nfiles```...)

Do you agree?

Reply:
Actually, they do not have to be the same length. The only requirement is all the EE data be filled before the EB. If you look into the calls to `getNNoutput` you will see that the last function parameter, `barrelstart` is either 0 or 1. In addition, all uses of `Nfiles_EE` in `getNNoutput` are multiplied by `barrelstart`.  Essentially `barrelstart*Nfiles_EE` sets to offset to the beginning of the stored EB info.

Reply:
Ok, thank you for re-checking.
I don't like that hard-coded "5", but I don't think it is worth insisting further on it: you already cleaned this class a lot!